kernel/memory/hexdump.c

```diff
@@ -0,0 +1,73 @@
+// SPDX-FileCopyrightText: CGL-KFS
+// SPDX-License-Identifier: BSD-3-Clause
+
+/* hexdump.c
+ *
+ * Exports a hexdump function to print memory
+ *
+ * created: 2022/11/19 - lfalkau <lfalkau@student.42.fr>
+ * updated: 2022/11/19 - lfalkau <lfalkau@student.42.fr>
+ */
+
+#include <kernel/vga.h>
+#include <kernel/print.h>
+#include <kernel/keyboard.h>
+
+static void wait_for_char_input(int c) {
+	struct kbd_event evt;
+
+	while (1) {
+		while (!KBD_poll());
+		KBD_getevent(&evt);
+		if (evt.type == KEY_PRESSED) {
+			if (c == 0 || KBD_getchar(&evt) == c)
+				break;
+		}
+	}
+}
+
+static int hexdump_diaplay_char(int c) {
+	if (c == 0) {
+		VGA_setforegroundcolor(VGA_COLOR_WHITE);
+		return '.';
+	} else if (c < 32 || c > 127) {
+		VGA_setforegroundcolor(VGA_COLOR_LIGHT_BLUE);
+		return '.';
+	} else {
+		VGA_setforegroundcolor(VGA_COLOR_LIGHT_GREEN);
+		return c;
+	}
+}
+
+/* Hexdump to print from address to limit
+ *
+ * @base is the address to start the print from
+ * @limit is the size we want to print from @base
+ *
+ */
+void hexdump(uint8_t *base, uint32_t size) {
```

Comment:
A minor change to the name of the variable in hexdump.
I think we could keep the conventional name limit (instead of size) to define the end of the print.
Otherwise this is a nice implementation and this hexdump function will help us in the future during a debug session.